src/Filter/AbstractDateFilter.php

```diff
@@ -119,26 +124,41 @@ public function getRenderSettings()
         ]];
     }
 
+    abstract protected function applyTypeIsLessEqual(ProxyQueryInterface $queryBuilder, string $field, array $data);
+
+    abstract protected function applyTypeIsGreaterThan(ProxyQueryInterface $queryBuilder, string $field, array $data);
+
+    abstract protected function applyTypeIsEqual(ProxyQueryInterface $queryBuilder, string $field, array $data);
+
     /**
      * @param string    $operation
      * @param string    $field
      * @param \DateTime $datetime
      */
-    protected function applyType(ProxyQueryInterface $queryBuilder, $operation, $field, \DateTime $datetime = null)
+    protected function applyType(ProxyQueryInterface $queryBuilder, $operation, $field, ?\DateTime $datetime = null)
     {
         $queryBuilder->field($field)->$operation($datetime);
         $this->active = true;
     }
 
     /**
+     * NEXT_MAJOR: Remove this method.
+     *
      * Returns if the filter type requires a value to be set.
      *
      * @param int $type
      *
+     * @deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
+     *
      * @return bool
      */
     protected function typeRequiresValue($type)
     {
+        @trigger_error(
+            'The '.__METHOD__.' is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
```

Comment:
```suggestion
            '"'.__METHOD__.'()" is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x and will be removed in version 4.0.'.
```